python_modules/dagster/dagster/_core/ext/utils.py

```diff
@@ -130,43 +130,43 @@ def read_messages(
         self,
         handler: "ExtMessageHandler",
     ) -> Iterator[ExtParams]:
-        with self.setup():
+        with self.setup() as params:
             is_task_complete = Event()
             thread = None
             try:
                 thread = Thread(
                     target=self._reader_thread,
                     args=(
                         handler,
+                        params,
                         is_task_complete,
                     ),
                     daemon=True,
                 )
                 thread.start()
-                yield self.get_params()
+                yield params
             finally:
                 is_task_complete.set()
                 if thread:
                     thread.join()
 
-    @contextmanager
-    def setup(self) -> Iterator[None]:
-        yield
-
     @abstractmethod
-    def get_params(self) -> ExtParams:
+    @contextmanager
+    def setup(self) -> Iterator[ExtParams]:
```

Comment:
please name `get_params` or `with_params` or something like that

Reply:
Renamed to `get_params`